quinn-udp/src/socket.rs

```diff
@@ -39,10 +37,6 @@ impl UdpSocket {
         Self::from_std(socket)
     }
 
-    pub fn socket_type(&self) -> SocketType {
-        self.ty
-    }
```

Comment:
sure we don't use this in libp2p-quic? you need to be able to figure out if you can dial ipv4 addresses when bound to an ipv6 address

Reply:
Ah, I see. I will revert this change if you are right